### PR TITLE
Add data mover interfaces for ceph backend

### PR DESCRIPTION
**Pull request by click2cloud-alpha**
@xing-yang As discussed this is tested for AWS to Ceph and vice-versa scenario for files ranging upto 2.4GB.

### CODE REVIEW

**Comment by xing-yang** [comment]
@sfzeng @wisererik Please take a look.

File: datamover/pkg/ceph/s3/s3mover.go [hunk 1]
```diff
@@ -0,0 +1,304 @@
+// Copyright (c) 2018 Huawei Technologies Co., Ltd. All Rights Reserved.
```
[the remark this answers is not in the document]
Comment (click2cloud-alpha) [reply datamover/pkg/ceph/s3/s3mover.go]: Addressed your comments.

**Comment by sfzeng** [comment]
Hi, I have two comments as below:
1. For the ListObjs of cephs3mover, filter seems not take effect.
2. Migration from ceph to azure blob and huaweicloud obs, and vice-versa also suggested to be tested.

**Comment by click2cloud-alpha** [comment]
Addressed your comments @sfzeng 
Apart from AWS S3, We have tested Azure Blob Storage as well.

**Comment by sfzeng** [comment]
@click2cloud-alpha good, thank you.

**Pull request merged**